projects/dev/src/app/group/group-manage/pending-request/pending-request.component.ts

```diff
@@ -1,108 +1,190 @@
+        severity: "warn",
+        summary: "Partial success",
+        detail: `${succ.true} request(s) have been ${msg}, ${succ.false} could not be executed`,
+        life: TOAST_LENGTH,
       });
     }
   }
 
-  constructor(private groupService: GroupService) {}
+  _processRequestError(_err) {
+    this.messageService.add({
+      severity: "error",
+      summary: "Error",
+      detail: ERROR_MESSAGE.fail,
+      life: TOAST_LENGTH,
+    });
+  }
+
+  isAccepting() {
+    return this.onGoingActivity === Activity.Accepting;
+  }
+
+  isRejecting() {
+    return this.onGoingActivity === Activity.Rejecting;
+  }
+
+  isIdle() {
+    return this.onGoingActivity === Activity.None;
+  }
+
+  constructor(
+    private groupService: GroupService,
+    private messageService: MessageService
+  ) {}
 
   ngOnInit() {
     this.panel.push({
       name: "Pending Requests",
       columns: this.columns,
     });
-    this.groupService
-      .getManagedRequests(this.id)
-      .subscribe((reqs: PendingRequest[]) => {
-        this._setRequestData(reqs);
-      });
   }
 
-  ngOnChanges(changes: SimpleChanges) {
-    this.groupService
-      .getManagedRequests(this.id)
-      .subscribe((reqs: PendingRequest[]) => {
-        this._setRequestData(reqs);
-      });
+  ngOnChanges(_changes: SimpleChanges) {
+    this.selection = [];
+    this._reloadData();
   }
 
-  onExpandWidth(e) {}
+  onAcceptOrReject(action: Action) {
+    if (this.selection.length === 0 || this.onGoingActivity !== Activity.None) {
+      return;
+    }
+
+    let resultObserver: Observable<RequestActionResponse>;
+    this.onGoingActivity = (action === Action.Accept) ? Activity.Accepting : Activity.Rejecting;
+    
+    const group_ids = this.selection.map((req: PendingRequest) => req.joining_user.group_id);
+
+    if (action === Action.Accept) {
+      resultObserver = this.groupService.acceptJoinRequest(this.id, group_ids);
+    } else {
+      resultObserver = this.groupService.rejectJoinRequest(this.id, group_ids);
+    }
 
-  onClickAccept(e) {
-    this.onAccept.emit(e);
+    resultObserver
+    .pipe(
+      tap((res: RequestActionResponse) => {
+        if (res.success === false || res.message !== "updated" || typeof res.data !== "object") {
+          throw "Unknown error";
+        }
+      })
+    )
+    .subscribe(
+      (res: RequestActionResponse) => {
+        this._displayResponseToast(
+          res,
+          action,
+          action === Action.Accept ? "accepted" : "declined"
+        );
+        this._reloadData();
+        this.onGoingActivity = Activity.None;
+        this.selection = [];
+      },
+      (err) => {
+        this._processRequestError(err);
+        this.onGoingActivity = Activity.None;
+      }
+    );
   }
 
-  onClickReject(e) {
-    this.onReject.emit(e);
+  onSelectAll(_event) {
+    if (this.selection.length === this.requests.length) {
+      this.selection = [];
+    } else {
+      this.selection = this.requests;
+    }
```

Comment:
I think this function can be completely removed (+`(click)="onSelectAll($event)"` in the html) and the behavior is still the same.

Reply:
Mathias asked me to make it work when I click on the text `Select All` so I made it so.
If it's an unnecessary behavior, I will remove it.

Reply:
Ah that's for the "select all" text, ok sorry.